aws_auth.tf

```diff
@@ -43,13 +43,10 @@ data "template_file" "worker_role_arns" {
 }
 
 data "template_file" "node_group_arns" {
-  count    = var.create_eks ? local.worker_group_managed_node_group_count : 0
+  count    = var.create_eks ? length(module.node_groups.aws_auth_roles) : 0
```

Comment:
Perhaps use a `for_each` loop here instead of count?

Reply:
Then you're just pushing the flattening of the map to further down the chain. Currently it's consistent with how other worker resources are generated.

I feel the whole aws-auth file generation could do with improvement but it's out of scope for this PR.